src/content/1.7/development/components/configuration.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Configuration storage
+weight: 80
+---
+
+# Configuration storage
+
+The Configuration storage of PrestaShop aims to address two needs.
+
+First, there is multiple situations where a single value has to be stored and be available easily. It can be a boolean to control order mode, it can be a limit for timeout: a simple key => value(s) storage is needed to manage multiple settings in PrestaShop.
+
+Secondly, modules might sometimes need a simple storage system, and they should not have to create a new SQL table for such a simple need.
+
+To address these needs, the Configuration storage of PrestaShop uses a SQL table `ps_configuration` and allows simple fetch, update and deletion of values in it.
+
+{{< figure src="/1.7/development/components/img/configuration_sql_structure.png" title="Configuration SQL structure" >}}
+
+Each value stored in the Configuration component can be controlled through given key, is compatible with Multishop (as it can be overriden for a shop or a group of shop) and has timestamps.
+
+The `Configuration` class is the interface allowing each module developer to use the component: to get or store data.
```

Comment:
We should prefer the usage of `PrestaShop\PrestaShop\Àdapter\Configuration` which is integrating the multishop behavior, no? :thinking:

Reply:
It depends if this documentation aims to explain "what is being used" or "what you should use" I guess

Maybe I can add a comment about this? "You will see the class Configuration used in many places but now please rather use the adapter" ?

Reply:
I think it's a good idea, but this Configuration is not working well with multistore context, we also have to talk about it.

Reply:
> I think it's a good idea, but this Configuration is not working well with multistore context, we also have to talk about it.

What do you mean? I thought it was one of the piece of multistore that was actually working quite correctly 😅  (compared to other stuff like BO Product page)

Reply:
Oh my bad, the multistore configuration is working well, but afaik, we have to prefer the usage of Adapter instead of native classes. That's the only point :sweat_smile:

Reply:
I agree with @PierreRambaud that we should encourage people to use an Adapter, especially that it's working fine with multi-store now

Reply:
maybe we need to have this as a separate paragraph? to inject adapter through constructor or create an instance of it, because there are no static methods in it

Reply:
We are supposed to encourage the use of interfaces, not adapters: remember, adapters are meant to go away ;)
...but that would require one interface for each class

Reply:
@eternoendless I was not precise enough, in almost every new code in PrestaShop, where we need, we're injecting @prestashop.adapter.legacy.configuration' 

it has a proper interface and... this is recommended way, right? 🤔

Reply:
Yep, that's the recommended way, depend on the interface, inject the adapter